src/lib/QtWidgets/QWidget/index.ts

```diff
@@ -152,12 +164,20 @@ export abstract class NodeWidget extends EventWidget {
   };
 }
 
-type Rect = {
-  x: number;
-  y: number;
+interface Size {
   width: number;
   height: number;
+}
+
+interface Point {
+  x: number;
+  y: number;
+}
+
+interface Rect extends Point, Size {
```

Comment:
Logically it doesnt make sense that Rect is extending from Point and Size.
Please create three different interfaces instead.